pkg/lockfile/parse-apk-installed.go

```diff
@@ -0,0 +1,99 @@
+package lockfile
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+const AlpineEcosystem Ecosystem = "Alpine"
+
+func groupApkPackageLines(scanner *bufio.Scanner) [][]string {
+	var groups [][]string
+	var group []string
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if line == "" {
+			if len(group) > 0 {
+				groups = append(groups, group)
+			}
+			group = make([]string, 0)
+			continue
+		}
+		group = append(group, line)
+	}
+
+	if len(group) > 0 {
+		groups = append(groups, group)
+	}
+
+	return groups
+}
+
+func parseApkPackageGroup(group []string, pathToLockfile string) PackageDetails {
+	var pkg = PackageDetails{}
+
+	pkg.Ecosystem = AlpineEcosystem
+	pkg.CompareAs = AlpineEcosystem
+
+	// File SPECS: https://wiki.alpinelinux.org/wiki/Apk_spec
+	for _, line := range group {
+		switch {
+		case strings.HasPrefix(line, "P:"):
+			pkg.Name = strings.TrimPrefix(line, "P:")
+		case strings.HasPrefix(line, "V:"):
+			pkg.Version = strings.TrimPrefix(line, "V:")
+		case strings.HasPrefix(line, "c:"):
+			pkg.Commit = strings.TrimPrefix(line, "c:")
+		}
+	}
+
+	if pkg.Version == "" {
+		_, _ = fmt.Fprintf(
+			os.Stderr,
+			"warning: malformed APK installed file. Found no version number in record. File: %s\n",
+			pathToLockfile,
+		)
+	}
+
+	return pkg
+}
+
+func ParseApkInstalled(pathToLockfile string) ([]PackageDetails, error) {
+
+	file, err := os.Open(pathToLockfile)
```

Comment:
nit: don't need this newline

```suggestion
func ParseApkInstalled(pathToLockfile string) ([]PackageDetails, error) {
	file, err := os.Open(pathToLockfile)
```